### PR TITLE
[Chips] Update MDCChipField to move long text from a line with chips to a new line.

### LINKED ISSUE #9006: Internal issue: b/144853451
This was filed as an internal issue. If you are a Googler, please visit [b/144853451](http://b/144853451) for more details.

---
#### Internal data
- Associated internal bug: [b/144853451](http://b/144853451)

### PR DESCRIPTION
**Pull request by bryanoltman**
Call layoutSubviews on textChanged to reposition MDCChipField's textField if:

1. The textField is on the same line as chips and becomes too long to fit
1. The textField is on the line below chips but can fit on the line with chips

Fixes #9006 

Before:
before 2020-03-03 15_08_50

After:
after 2020-03-03 15_10_39

### CODE REVIEW

File: components/Chips/src/MDCChipField.m
Comment (jverkoey) [inline components/Chips/src/MDCChipField.m:632]: I'm finding it hard to identify what's net-new in this PR vs what's a refactoring; is it possible to split this into two smaller PRs?

**Review by jverkoey — APPROVED** [review]
I have some UX #opinions on this that I'm ok with deferring to follow-up, namely around the fact that the label visually jumps while being edited. I'd prefer if the input field was always on a new line so that it stayed in a consistent visual anchoring point at all times during input of a single chip.

Otherwise, overall the UX LGTM as an iterative improvement.

File: components/Chips/src/MDCChipField.m
Comment (bryanoltman) [reply components/Chips/src/MDCChipField.m:632]: Pulled the refactoring into https://github.com/material-components/material-components-ios/pull/9827

**Comment by jverkoey** [comment]
Strange...I had re-requested review on this PR. Did the merge button go green even without my approval bit being green?

**Comment by bryanoltman** [comment]
Hmm, yeah, it looked like I still had the approval, although looking at the history for this PR, you definitely self-requested a review. Does self-requesting not dismiss existing reviews? I would sort of expect it to.

**Comment by jverkoey** [comment]
No worries, I understand how it happened. We should look into why GitHub didn’t flag the PR for needing review though.

**Pull request merged**
